src/gnssro/gnssro_bufr2ioda.py

```diff
@@ -227,12 +235,20 @@ def get_obs_data(bufr, profile_meta_data, add_qc, addLSW, record_number=None):
     obs_data[('bendingAngle', "PreQC")] = np.full(krepfac[0], 0, dtype=ioda_int_type)
 
     # (geometric) height is read as integer but expected as float in output
-    height = codes_get_array(bufr, 'height', ktype=float)
+    if only_bang:
+        refrac = np.empty_like(bang)
+        refrac[:] = float_missing_value
+        refrac_err = np.empty_like(bang_err)
+        refrac_err[:] = float_missing_value
+    else:
+        # get the refractivity
+        refrac = codes_get_array(bufr, 'atmosphericRefractivity')[0::2]
+        refrac_err = codes_get_array(bufr, 'atmosphericRefractivity')[1::2]
 
-    # get the refractivity
-    refrac = codes_get_array(bufr, 'atmosphericRefractivity')[0::2]
-    refrac_err = codes_get_array(bufr, 'atmosphericRefractivity')[1::2]
-    refrac_conf = codes_get_array(bufr, 'percentConfidence')[sum(krepfac[:1])+1:sum(krepfac[:2])+1]
+    try:
+        height = codes_get_array(bufr, 'geopotentialHeight', ktype=float)
+    except Exception as e:
+        height = codes_get_array(bufr, 'height', ktype=float)
 
```

Comment:
height = codes_get_array(bufr, 'geopotentialHeight', ktype=float)

This line caused the failure of the ctest failure: test_iodaconv_generic_gnssro_bufr
with the PR code, _height_ becomes _geopotentialHeight_, which differs from the reference height. @BenjaminRuston this is something we need to define. Height refers to geometric height usually in RO. If we want geopentialHeight, we can add an additional variable for this. Thoughts?

Reply:
@huishao-r Seems we should keep both or get rid of one of them. Correct?

Reply:
@huishao-r The code as is reads in the geopotentialHeight only if it is present and ignores the height.

Reply:
@huishao-r @BenjaminRuston @fcvdb I switched the order above and the test now pass. Can you take a look and see if this is just fluke?

Reply:
@huishao-r @fcvdb What I am wondering is if an exception is actually thrown if the variable is not present in the bufr file. That would require some extra processing such a see if the size of the numpy array is zero and then throw your own exception.